File: tiq-e/others/4/main.go
package main

import (
	"fmt"
	"time"
)

func generate(numRows int) [][]int {
	var answer [][]int
	if numRows == 0 {
		return answer
	}
	answer = append(answer, []int{1})
	for i := 1; i < numRows; i++ {

		row := make([]int, i+1)
		row[0] = 1
		row[len(row)-1] = 1

		for j := 1; j < len(row)-1; j++ {
			row[j] = answer[i-1][j-1] + answer[i-1][j]
		}
		answer = append(answer, row)
	}
	return answer
}

func main() {
	x := 5

	start := time.Now()
	result := generate(x)
	end := time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}

File: tiq-e/others/4/main.ts
import {performance} from "perf_hooks";

function generate(numRows: number): number[][] {
    if (numRows === 0) return [];

    let rows: number[][] = [[1]];

    for (let i = 1; i < numRows; i++) {
        let newRow = [1];
        for (let j = 1; j <= i; j++)
            newRow.push(rows[i - 1][j - 1] + (rows[i - 1][j] || 0));
        rows.push(newRow);
    }
    return rows;
}

const x = 5;

const start = performance.now();
const result = generate(x);
const end = performance.now();
console.log("Duration:", end - start);
console.log("Result:", result);
